docs: add upgrading doc for v1 release (#2832)
googleapis/genai-toolbox: UPGRADING.md

# Upgrading to MCP Toolbox for Databases v1.0.0

Welcome to the v1.0.0 release of the MCP Toolbox for Databases! 

This release stabilizes our core APIs and standardizes our protocol alignments.
As part of this milestone, we have introduced several breaking changes and
deprecations that require updates to your configuration and code.

**📖 New Versioning Policy**
We have officially published our [Versioning Policy](https://googleapis.github.io/genai-toolbox/dev/about/versioning/). Moving forward, we follow standard versioning conventions to classify updates:
* **Major (vX.0.0):** Breaking changes requiring manual updates.
* **Minor (v1.X.0):** New, backward-compatible features and deprecation notices.
* **Patch (v1.0.X):** Backward-compatible bug fixes and security patches.

This guide outlines what has changed and the steps you need to take to upgrade.

## 🚨 Breaking Changes (Action Required)

### 1. Endpoint Standardization: `/api` removed

> The legacy `/api` endpoint has been completely removed. All official SDKs have been updated to use the `/mcp` endpoint instead.

Reviewer: ```suggestion
The legacy `/api` endpoint for the native Toolbox protocol has been completely removed. All official SDKs have been updated to use the `/mcp` endpoint, which now aligns with the standard Model Context Protocol (MCP) specification.
```

* **Migration:** You must update all your implementation to use the `/mcp`

>   endpoint exclusively.

Reviewer: ```suggestion
* **Migration:** You must update all custom implementations to use the `/mcp`
  endpoint exclusively. We want to ensure no functionality is lost in this transition. If your workflow  
  relied on a non-standard feature that is missing from the new implementation, please submit a
  feature request on our [GitHub Issues page](https://github.com/googleapis/genai-toolbox/issues).
```

### 2. Strict Tool Naming Validation (SEP986)
Tool names are now strictly validated against [ModelContextProtocol SEP986 guidelines](https://github.com/alexhancock/modelcontextprotocol/blob/main/docs/specification/draft/server/tools.mdx#tool-names) prior to MCP initialization.
* **Migration:** Ensure all your tool names **only** contain alphanumeric characters, hyphens (`-`), underscores (`_`), and periods (`.`). Any other special characters will cause initialization to fail.

### 3. Removed CLI Flags
The legacy snake_case flag `--tools_file` has been completely removed.
* **Migration:** Update your deployment scripts to use `--config` instead.

> ### 4. Singular `kind` Values in Configuration

Reviewer: Can you add a first sentence caveat like "if you have migrated to the Flat Configuration Format" and link to the item below too?

used to declare the core primitive of the block (e.g., `source` or `tool`).

**Example of the new flat format:**

```yaml
kind: source
name: my-source
type: alloydb-postgres
project: my-project
region: my-region
instance: my-instance
---
kind: tool
name: my-simple-tool
type: postgres-execute-sql
source: my-source
description: this is a tool that executes the sql provided.
```

**Migration:**

You can automatically migrate your existing nested configurations to the new flat format using the CLI. Run the following command:

```Bash
./toolbox migrate --config <path-to-your-config>
```
_Note: You can also use the `--configs` or `--config-folder` flags with this command._

### 2. Deprecated CLI Flags
The following CLI flags are deprecated and will be removed in a future release. Please update your scripts:

* `--tools-file` ➡️ Use `--config`
* `--tools-files` ➡️ Use `--configs`
* `--tools-folder` ➡️ Use `--config-folder`

## 💡 Other Notable Updates
* **Enhanced Error Handling:** Errors are now strictly categorized between Agent Errors (allowing the LLM to self-correct) and Client/Server Errors (which signal a hard stop).

> * **Telemetry Updates:** The /mcp endpoint telemetry has been revised to fully comply with the OpenTelemetry semantic conventions for MCP.

Reviewer: ```suggestion
* **Telemetry Updates:** The /mcp endpoint telemetry has been revised to fully comply with the [OpenTelemetry semantic conventions for MCP](https://opentelemetry.io/docs/specs/semconv/gen-ai/mcp/).
```

> * **MCP Auth Support:** The ModelContextProtocol's auth specification is now fully supported.

Reviewer: ```suggestion
* **MCP Authorization Support:** The Model Context Protocol's [authorization specification](https://modelcontextprotocol.io/specification/2025-11-25/basic/authorization) is now fully supported.
```

* **Database Name Validation:** Removed the "required field" validation for the database name in CloudSQL for MySQL and generic MySQL sources.

> * **Prebuilt Tools:** Toolsets have been resized for better performance, and tool names are now aligned with OneMCP.

Reviewer: ```suggestion
* **Prebuilt Tools:** Toolsets have been resized for better performance.
```
